test/integration_test/kubevirt_test.go

```diff
@@ -246,3 +251,67 @@ func createDatadiskTemplates(t *testing.T) error {
 	}
 	return nil
 }
+
+func kubevirtVMScaledDeployAndValidate(t *testing.T, instanceID, appKey string, scale int) []*scheduler.Context {
+	var wg sync.WaitGroup
+	var allCtxs []*scheduler.Context
+	logrus.Infof("Deploying %d VMs and validating them.", scale)
+	for i := 1; i <= scale; i++ {
+		logrus.Infof("Deploying VM: %d", i)
+		wg.Add(1)
+		ctxs := kubevirtVMsDeployAndValidate(
+			t,
+			instanceID+"-"+fmt.Sprintf("%d", i),
+			[]string{appKey},
+			&wg,
+		)
+		logrus.Infof("Validated VM: %d", i)
+		allCtxs = append(allCtxs, ctxs...)
+	}
+	wg.Wait()
+	return allCtxs
+}
+
+func createClusterroleDataVolume(clusterroleName string) error {
+	clusterRoleDatavolume := &rbacv1.ClusterRole{
+		ObjectMeta: meta.ObjectMeta{
+			Name: clusterroleName,
+		},
+		Rules: []rbacv1.PolicyRule{{
+			Verbs:     []string{"*"},
+			APIGroups: []string{"cdi.kubevirt.io"},
+			Resources: []string{"*"},
```

Comment:
Can we tighten it further and add only the verbs and resources that were missing as per the error message? I believe all we need is "clone" permission.

Reply:
Tightened the role by adding the following resources `datavolumes/source` but for verbs we do need `*` for cloning. 
Refer this document: https://access.redhat.com/documentation/en-us/openshift_container_platform/4.14/html/virtualization/storage#virt-enabling-user-permissions-to-clone-datavolumes